packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -99,10 +123,10 @@ export class AttributionCollection<T> {
         let currentInfo = keys[curIndex];
 
         for (const segment of segments) {
-            const attribution = new AttributionCollection(currentInfo, segment.cachedLength);
+            const attribution = new AttributionCollection(currentInfo as number, segment.cachedLength);
```

Comment:
outside the loop i think we should `assert(typeof currentInfo === "number",...)` rather than the casts, which will silently break at runtime

Reply:
i made the typing on the summary more strict to only take numbers now, which aligns better with the re-typing of `AttributionKey` as `{ type: "op"; seq: number }`. so no asserts necessary.